source/hardware/controllers/pioneer_ddj_400.rst

```diff
@@ -0,0 +1,319 @@
+
+Pad Modes
+^^^^^^^^^
+
+The pad mode buttons change between different modes in which the performance pads operate. The main
+pad modes described below are similar to those found in rekordbox.
+
+.. note:: The secondary pad modes described in the manufacturer's manual
+          (:hwlabel:`KEYBOARD` :hwlabel:`PAD FX1` :hwlabel:`PAD FX2` and
+          :hwlabel:`KEYSHIFT`) are not currently implemented by this
+          mapping.
+
+Hot Cue Mode
+^^^^^^^^^^^^
+.. _pioneer-ddj-400-hot-cue-mode:
+
+Bookmark positions in the track and jump to them.
+
+.. figure:: ../../_static/controllers/pioneer_ddj_400_performancepads.svg
+   :align: center
+   :width: 40%
+   :figwidth: 100%
+   :alt: Pioneer DDJ-400 (hot cue pads)
+   :figclass: pretty-figures
+
+   Pioneer DDJ-400 (hot cue pads)
+
+
+========  ===============================  ========================================================
+No.       Control                          Function
+========  ===============================  ========================================================
+1-8       Pad (unlit)                      Save current position as hot cue.
+1-8       Pad (lit)                        Jump to hot cue. If the track is stopped, holding the
+                                           pad will preview the hot cue until the pad is released.
+1-8       :hwlabel:`SHIFT` + Pad (lit)     Clear hot cue.
+========  ===============================  ========================================================
+
+Beat Loop Mode
+^^^^^^^^^^^^^^
+.. _pioneer-ddj-400-beat-loop-mode:
+
+Press a pad to enable a loop from the current position. Press the pad again to
+disable the loop. The size of the loop is assigned to the pads as follows:
+
+.. figure:: ../../_static/controllers/pioneer_ddj_400_performancepads.svg
+   :align: center
+   :width: 40%
+   :figwidth: 100%
+   :alt: Pioneer DDJ-400 (beat loop pads)
+   :figclass: pretty-figures
+
+.. table:: Pioneer DDJ-400 (beat loop sizes)
+   :align: center
+
+   ============  ============  ============  ============
+   1/4 beat      1/2 beat      1 beat        2 beats
+   4 beats       8 beats       16 beats      32 beats
+   ============  ============  ============  ============
+
+Beat Jump Mode
+^^^^^^^^^^^^^^
+.. _pioneer-ddj-400-beat-jump-mode:
+
+Press a pad to jump forwards or backwards by the number of beats assigned to the
+pads below.
+
+Press :hwlabel:`SHIFT` + pad 7 to decrease or :hwlabel:`SHIFT` + pad 8 to
+increase the jump values by a factor of 16.
+
+.. figure:: ../../_static/controllers/pioneer_ddj_400_performancepads.svg
+   :align: center
+   :width: 40%
+   :figwidth: 100%
+   :alt: Pioneer DDJ-400 (beat jump pads)
+   :figclass: pretty-figures
+
+.. table:: Pioneer DDJ-400 (beat jump defaults)
+   :align: center
+
+   ====  =======  =========
+   No.   Jump     Direction
+   ====  =======  =========
```

Comment:
I think a 2x4 table matching the layout of the pads would be easier to understand here.